metrics/tracing/util.go

```diff
@@ -0,0 +1,16 @@
+package tracing
+
+import (
+	"context"
+
+	"go.opencensus.io/trace"
+)
+
+// AddErrorEndSpan will end `span` and adds `err` to `span` iff err is not nil.
+// This is a helper method to cut down on boiler plate.
+func AddErrorEndSpan(ctx context.Context, span *trace.Span, err *error) {
+	if *err != nil {
+		span.AddAttributes(trace.StringAttribute("error", (*err).Error()))
+	}
+	span.End()
+}
```

Comment:
👍 removing boilerplate. You could also add a `StartSpan` with varargs to inline string attributes.

Reply:
I agree there is a lot of boilerplate with this. However, I would prefer to leave this as is until a clearer pattern emerges, I will address it in a follow on at that point.